azure-pipelines.yml

```diff
@@ -6,77 +6,95 @@
+            $ANDROID_HOME/platform-tools/adb devices
 
-      $ANDROID_HOME/platform-tools/adb devices
+            $ANDROID_HOME/platform-tools/adb shell settings put global window_animation_scale 0 &
+            $ANDROID_HOME/platform-tools/adb shell settings put global transition_animation_scale 0 &
+            $ANDROID_HOME/platform-tools/adb shell settings put global animator_duration_scale 0 &
 
-      $ANDROID_HOME/platform-tools/adb shell settings put global window_animation_scale 0 &
-      $ANDROID_HOME/platform-tools/adb shell settings put global transition_animation_scale 0 &
-      $ANDROID_HOME/platform-tools/adb shell settings put global animator_duration_scale 0 &
+            echo "Emulator started"
+          displayName: 'Run Android Emulator'
 
-      echo "Emulator started"
-    displayName: 'Run Android Emulator'
+        - bash: |
+            $ANDROID_HOME/platform-tools/adb shell """
+            mount -o rw,remount rootfs /;
+            chmod 0777 /mnt/sdcard;
+            exit
+            """
+            sleep 10
+            $ANDROID_HOME/platform-tools/adb logcat -c
+            $ANDROID_HOME/platform-tools/adb logcat > $(Build.ArtifactStagingDirectory)/device.log &
+            echo $! > logcat.pid
+            echo "Logcat Started with Process ID: $(cat logcat.pid)"
+            $ANDROID_HOME/platform-tools/adb shell "mkdir -p $(DEVICE_VIDEO_DIR)"
+            $ANDROID_HOME/platform-tools/adb shell '''
+              for i in $(seq 1 1 20)
+              do
+                screenrecord --bugreport --verbose --time-limit=180 /mnt/sdcard/DCIM/test"${i}".mp4
+              done
+            ''' > $(Build.ArtifactStagingDirectory)/video.log &
+            echo $! > video.pid
+            echo "Video Recorder Started with Process ID: $(cat video.pid)"
+          displayName: 'Start Video Record and Logcat'
 
-  - bash: |
-      $ANDROID_HOME/platform-tools/adb shell """
-      mount -o rw,remount rootfs /;
-      chmod 0777 /mnt/sdcard;
-      exit
-      """
-      sleep 10
-      $ANDROID_HOME/platform-tools/adb logcat -c
-      $ANDROID_HOME/platform-tools/adb logcat > $(Build.ArtifactStagingDirectory)/device.log &
-      echo $! > logcat.pid
-      echo "Logcat Started with Process ID: $(cat logcat.pid)"
-      $ANDROID_HOME/platform-tools/adb shell "mkdir -p $(DEVICE_VIDEO_DIR)"
-      $ANDROID_HOME/platform-tools/adb shell '''
-        for i in $(seq 1 1 20)
-        do
-          screenrecord --bugreport --verbose --time-limit=180 /mnt/sdcard/DCIM/test"${i}".mp4
-        done
-      ''' > $(Build.ArtifactStagingDirectory)/video.log &
-      echo $! > video.pid
-      echo "Video Recorder Started with Process ID: $(cat video.pid)"
-    displayName: 'Start Video Record and Logcat'
+        - bash: |
+            ./gradlew connectedDebugAndroidTest --info
```

Comment:
Why Debug?

Reply:
I've tried release but it seems to require signing:

`Unable to install /Users/runner/work/1/s/app/build/outputs/apk/release/app-release-unsigned.apk
com.android.ddmlib.InstallException: INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed to collect certificates from /data/app/vmdl180142812.tmp/base.apk: Attempt to get length of null array
`

[Link to failed build](https://dev.azure.com/warpedomen/warpedomen/_build/results?buildId=38&view=logs&j=aa2fd561-a338-5537-45ee-49ae22aea958&t=e99b6747-5a34-5948-7b31-f1617fea6ef1&l=34065)

Anyway it doesn't set up in the pipeline, we should add build type in gradle - I've opened issue for it https://github.com/omen273/crosswordApp/issues/52